src/couch_replicator_doc_processor.erl

```diff
@@ -88,6 +89,16 @@ db_change(DbName, {ChangeProps} = Change, Server) ->
     Server.
 
 
+-spec get_worker_ref(db_doc_id()) -> reference() | nil.
+get_worker_ref({DbName, DocId}) when is_binary(DbName), is_binary(DocId) ->
+    case ets:lookup(?MODULE, {DbName, DocId}) of
+        [#rdoc{worker = WRef}] when is_reference(WRef) ->
+            WRef;
+        _Other ->
```

Comment:
Good point. You're right. We expect both. [] in case document was deleted already. Or an #rdoc with a different worker reference. I will make both case explicit